### ISSUE TITLE
Explicit delegate construction doesn't warn on nullable mismatch

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#34731]
**Version Used**: 
3.1.0-beta1-19172-05+edd2de88fb3e84a097fb30b4070e0f219f624e40

**Steps to Reproduce**:
```C#
#nullable enable
using System;

public class Program
{
    public static void Main()
    {
        Action<object?> first = Method;
        Action<object?> second = new Action<object?>(Method);
    }

    private static void Method(object state) { }
}
```

There is warning on `first` ("Nullability of reference types in type of parameter 'state' of 'void Program.Method(object state)' doesn't match the target delegate 'Action<object?>'") but not on `second`.

![image](https://user-images.githubusercontent.com/2642209/55506239-ee653e00-5622-11e9-8ad2-99572a8e27cc.png)

cc: @cston, @jaredpar, @dotnet/nullablefc

**Comment by svick** [comment]
This looks very similar to https://github.com/dotnet/roslyn/issues/32499.

**Comment by stephentoub** [comment]
Thanks, @svick.  My searching skills failed me on that one; yes, this looks like a dup of that.
